src/TLuaInterpreter.cpp

```diff
@@ -5786,6 +5787,291 @@ int TLuaInterpreter::purgeMediaCache(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#sourceMedia
+int TLuaInterpreter::sourceMedia(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    TMediaData mediaData{};
+
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "sourceMedia: bad argument #1 type (message as string expected, got %1!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    mediaData.setMediaProtocol(TMediaData::MediaProtocolAPI);
+    mediaData.setMediaUrl(lua_tostring(L, -1));
+    host.mpMedia->sourceMedia(mediaData);
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#loadMedia
+int TLuaInterpreter::loadMedia(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    TMediaData mediaData{};
+
+    if (!lua_istable(L, 1)) {
+        lua_pushfstring(L, "loadMedia: bad argument #1 type (commands as table expected, got %1!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+    lua_pushnil(L);
+    while (lua_next(L, 1) != 0) {
+        // key at index -2 and value at index -1
+        if (lua_type(L, -2) != LUA_TSTRING) {
+            lua_pushfstring(L, "loadMedia: bad argument #1 type (table keys must be strings, got index: %s (must be string))", luaL_typename(L, -2));
+            return lua_error(L);
+        }
+
+        QString key = lua_tostring(L, -2);
+        key = key.toLower();
+
+        if (key == QLatin1String("name") || key == QLatin1String("key") || key == QLatin1String("tag")) {
+            if (!lua_isstring(L, -1)) {
+                lua_pushfstring(L, "loadMedia: bad argument #1 type (for index %s, table values must be strings, got value: %s (must be string))", luaL_typename(L, -2), luaL_typename(L, -1));
+                return lua_error(L);
+            }
+
+            QString value = lua_tostring(L, -1);
+
+            if (key == QLatin1String("name") && !value.isEmpty()) {
+                mediaData.setMediaFileName(value);
+            } else if (key == QLatin1String("key") && !value.isEmpty()) {
+                mediaData.setMediaKey(value);
+            } else if (key == QLatin1String("tag") && !value.isEmpty()) {
+                mediaData.setMediaTag(value);
+            }
+        }
+
+        // removes value, but keeps key for next iteration
+        lua_pop(L, 1);
+    }
+
+    if (mediaData.getMediaFileName().isEmpty()) {
+        return warnArgumentValue(L, __func__, "loadMedia: missing required information for \"name\" (add index \"name\" and value of file to play)");
```

Comment:
`warnArgumentValue` already prints the function name, so we can skip that. Suggest simplifying the suggestion to lower technical bar as wel:

```suggestion
        return warnArgumentValue(L, __func__, R"(missing name (add "name" = "file to play"))");
```

Reply:
Shortened this a good bit to eliminate the quoting.  The `R` syntax fools with my vscode a bit, so I avoided it for now and just kept this very simple.